x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/pivot/AggregationResultUtils.java

```diff
@@ -46,6 +46,7 @@
                                                                                  DataFrameIndexerTransformStats stats) {
         return agg.getBuckets().stream().map(bucket -> {
             stats.incrementNumDocuments(bucket.getDocCount());
+            stats.incrementCurrentRunDocsProcessed(bucket.getDocCount());
```

Comment:
I also considered trying to figure out a way to increment once the docs are indexed...but that seemed too complicated with very little benefit. The only downside is that we could hit 100% docs being pulled, but still have to index them into destination index, so the progress is not really "DONE". However, I think the indexer state that we return to the user might be enough to show that it is not "fully complete" and still needs to finish indexing the documents.